tools/tenant_cleaner.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2014 Juniper Networks, Inc. All rights reserved.
+#
+
+import argparse
+import json
+import re
+import requests
+import sys
+import os
+
+from keystoneclient.v2_0 import client as ksclient
+from novaclient import client as novaclient
+
+
+class Tenant(object):
+
+    def __init__(self, token=None, server_ip='127.0.0.1',
+                 port=8082,
+                 username=os.environ.get('OS_USERNAME', 'admin'),
+                 password=os.environ.get('OS_PASSWORD', 'admin'),
+                 auth_url=os.environ.get('OS_AUTH_URL', 'http://127.0.0.1:5000/v2.0/'),
+                 auth_tenant=os.environ.get('OS_TENANT', 'admin')
+                 ):
+        self._server_ip = server_ip
+        self._server_port = port
+        self.username = username
+        self.password = password
+        self.auth_url = auth_url
+        self.auth_tenant = auth_tenant
+        self.available_tenants = None
+
+        if token:
+            self._token = token
+        else:
+            self._token = self.get_keystone_token()
+
+        self.headers = {'X-Auth-Token': self._token}
+
+    def cleanup(self, tenants):
+        for tenant in tenants:
+            nova_objects_deleted = self.cleanup_nova_objects(tenant)
```

Comment:
Lets do this..
After we try deleting the vms, lets have a check to make sure all vms are deleted (with retries)
If they are not deleted at the end, we should just exit with a negative return value. Let the user take care of it himself